src/core/vectortile/qgsmapboxglstyleconverter.cpp

```diff
@@ -625,6 +629,225 @@ bool QgsMapBoxGlStyleConverter::parseLineLayer( const QVariantMap &jsonLayer, Qg
+        break;
+
+      case QVariant::String:
+        circleStrokeColor = parseColor( jsonCircleStrokeColor.toString(), context );
+        break;
+
+      default:
+        context.pushWarning( QObject::tr( "%1: Skipping unsupported circle-stroke-color type (%2)" ).arg( context.layerId(), QMetaType::typeName( jsonCircleStrokeColor.type() ) ) );
+        break;
+    }
+  }
+
+  // circle stroke width
+  double circleStrokeWidth = -1.0;
+  if ( jsonPaint.contains( QStringLiteral( "circle-stroke-width" ) ) )
+  {
+    const QVariant circleStrokeWidthJson = jsonPaint.value( QStringLiteral( "circle-stroke-width" ) );
+    switch ( circleStrokeWidthJson.type() )
+    {
+      case QVariant::Int:
+      case QVariant::Double:
+        circleStrokeWidth = circleStrokeWidthJson.toDouble() * context.pixelSizeConversionFactor();
+        break;
+
+      case QVariant::Map:
+        circleStrokeWidth = -1.0;
+        ddProperties.setProperty( QgsSymbolLayer::PropertyStrokeWidth, parseInterpolateByZoom( circleStrokeWidthJson.toMap(), context, context.pixelSizeConversionFactor(), &circleStrokeWidth ) );
+        break;
+
+      case QVariant::List:
+      case QVariant::StringList:
+        ddProperties.setProperty( QgsSymbolLayer::PropertyStrokeWidth, parseValueList( circleStrokeWidthJson.toList(), PropertyType::Numeric, context, context.pixelSizeConversionFactor(), 255, nullptr, &circleStrokeWidth ) );
+        break;
+
+      default:
+        context.pushWarning( QObject::tr( "%1: Skipping unsupported circle-stroke-width type (%2)" ).arg( context.layerId(), QMetaType::typeName( circleStrokeWidthJson.type() ) ) );
+        break;
+    }
+  }
+
+  double circleStrokeOpacity = -1.0;
+  if ( jsonPaint.contains( QStringLiteral( "circle-stroke-opacity" ) ) )
+  {
+    const QVariant jsonCircleStrokeOpacity = jsonPaint.value( QStringLiteral( "circle-stroke-opacity" ) );
+    switch ( jsonCircleStrokeOpacity.type() )
+    {
+      case QVariant::Int:
+      case QVariant::Double:
+        circleStrokeOpacity = jsonCircleStrokeOpacity.toDouble();
+        break;
+
+      case QVariant::Map:
+        ddProperties.setProperty( QgsSymbolLayer::PropertyStrokeColor, parseInterpolateOpacityByZoom( jsonCircleStrokeOpacity.toMap(), circleStrokeColor.isValid() ? circleStrokeColor.alpha() : 255 ) );
+        break;
+
+      case QVariant::List:
+      case QVariant::StringList:
+        ddProperties.setProperty( QgsSymbolLayer::PropertyStrokeColor, parseValueList( jsonCircleStrokeOpacity.toList(), PropertyType::Opacity, context, 1, circleStrokeColor.isValid() ? circleStrokeColor.alpha() : 255 ) );
+        break;
+
+      default:
+        context.pushWarning( QObject::tr( "%1: Skipping unsupported circle-stroke-opacity type (%2)" ).arg( context.layerId(), QMetaType::typeName( jsonCircleStrokeOpacity.type() ) ) );
+        break;
+    }
+  }
+
```

Comment:
It should also be possible to convert circle-translate to an offset on the symbol -- see eg how fill-translate is converted.